File: LeetCode/[E] - 643. Maximum Average Subarray I/solution.go
package main

func findMaxAverage(nums []int, k int) float64 {
	curr := 0
	for i := 0; i < k; i++ {
		curr += nums[i]
	}
	res := curr
	for r := k; r < len(nums); r++ {
		curr += nums[r] - nums[r-k]
		res = max(res, curr)
	}
	return float64(res) / float64(k)
}

File: LeetCode/[E] - 643. Maximum Average Subarray I/solution.py
from typing import List


class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        wind = sum(nums[:k])
        res = wind
        l = 0
        for r in range(k, len(nums)):
            wind += nums[r] - nums[l]
            res = max(res, wind)
            l += 1
        return res / k
